### ISSUE TITLE
Hide sensitive data in Survey Variable when user types into that field

### ISSUE THREAD

**Issue by UNiXMIT** [ansible-semaphore/semaphore#1196]
In my survey variables I enter my credentials that are used to authenticate with SSO in my playbooks so that the AWS connection succeeds. Is it possible to mark a survey variable as sensitive so that passwords are masked by ***** when entering the data/password? Like there is an option to use 'required="required"' can there also be an option to add 'type="password"' i.e.
![image](https://user-images.githubusercontent.com/4117073/226870008-6fc2d2e8-eec5-4145-9a6e-b507225c3102.png)

![image](https://user-images.githubusercontent.com/4117073/226869866-53a97dbb-275b-43f4-b3bf-08b84bb01485.png)

FYI I applied 'type="password"' using the browser developer tools.

Thanks

**Comment by UNiXMIT** [comment]
As a workaround I am injecting the following Javascript into the page:

```
let observer = new MutationObserver(mutations => {
	for (const elem of document.querySelectorAll("label")) {
	  if (elem.textContent.includes("SSO Password")) {
	    elem.nextSibling.type = 'password';
	  }
	}
});
observer.observe(document, {childList: true, subtree: true});
```

It might also be useful to add other input types on fields like email etc.

Thanks and keep up the great work.

**Comment by UNiXMIT** [comment]
I've been able to improve the workaround with the same javascript but making nginx add the script header in the html to load the javascript file. No need for a browser extension now.

**Comment by relwell** [comment]
I agree that there should be a way to mask survey variables. Sometimes we want to parameterize one-off things that should still be treated as secrets. Even just creating a "sensitive-string" type that's different from the current "secret" type (which doesn't seem to work for one-offs)

**Comment by fiftin** [comment]
Hi @relwell I'm sorry I can't understand what the problem with Secret?

**Comment by relwell** [comment]
@fiftin when we create fields in our templates, we should be able specify whether or not the field contains sensitive data (but is not a password, for instance), and in those cases the field should be masked the same way. In other words, we shouldn't use the password field for things that are not passwords, but we still need a way to mask the input in the UI.
